db/dcrpg/sync.go

```diff
@@ -218,9 +229,15 @@ func (db *ChainDB) SyncChainDB(client rpcutils.MasterBlockGetter, quit chan stru
 		if nodeHeight, err = client.NodeHeight(); err != nil {
 			return ib, fmt.Errorf("GetBestBlock failed: %v", err)
 		}
+
+		// if updating explorer is activated, update it at intervals of 20
+		if ib%20 == 0 && explorer.SyncExplorerUpdateStatus() && updateExplorer != nil && !updateAllAddresses {
```

Comment:
`explorer.SyncExplorerUpdateStatus` might still be important since the channel `updateExplorer` might not be nil but the update should not go through.

Reply:
Sure, you keep all conditions that are needed, but by putting `updateExplorer != nil` first it won't bother to call the function if `updateExplorer==nil`